### DISCUSSION TITLE
Tasmota Bluetooth Classic communication

### DISCUSSION THREAD

**Discussion by ramanraja** [arendst/Tasmota#18825] [Q&A] [answered]
I want to send a string to Tasmota on ESP-32 through its Bluetooth classic interface. On the sending side, I have the mobile application Serial Bluetooth Terminal. But how can I receive that string on the Tasmota side? My intent is to give the WiFi credentials to Tasmota through Bluetooth. I may probably have to write a Tasmota rule or Berry script etc., but I need a concrete example.

**Comment by Jason2866** [comment]
Bluetooth classic is NOT supported.

**Reply by ramanraja** [reply]
OK, I see. Is it possible to get it done with BLE ? I just want to send a single string from a mobile to the Tasmota device.

**Reply by Staars** [reply, marked as answer]
Yes this is possible, but not in an official build.
This example shows a mix of Improv-via-BLE and simple writing to a characteristic, which can be used to execute any Tasmota command via a virtual BLE console.

https://github.com/arendst/Tasmota/pull/18494#issuecomment-1520517816

As always with Bluetooth on any platform, you have to code both endpoints of the „line“.
